test/e2e/functional_test.go

```diff
@@ -0,0 +1,368 @@
+package prometheus
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
```

Comment:
We don't use ginkgo in this project, do you think we can avoid importing this and just follow the style of the e2e tests we already have. See the other files for inspiration:
https://github.com/openshift/cluster-monitoring-operator/blob/master/test/e2e/telemeter_test.go

Reply:
Thank you for review. 
First of all was set a task for me write tests using ginkgo framework. So after I wrote some tests  I meet the problem where it should be committed, because first place `extended/prometheus` was a wrong place :(.  I think I could be write in unit test style but I can not figure out workflow of e2e tests. Could not you explain me please? p.s Might be you have some gdoc with describing how run unit tests on laptop, do debug etc.

Reply:
No problem, sorry about the initial confusion!

So about these tests, they are still end-to-end tests, to run them locally you need an openshift cluster and you just run them using the Makefile target `make e2e` https://github.com/openshift/cluster-monitoring-operator/blob/master/Makefile#L132-L134 the `--kubeconfig` flag takes in the path to your KUBECONFIG file which should contain the openshift cluster config.

That should run all the e2e tests we have currently. The setup we have currently uses the https://github.com/openshift/cluster-monitoring-operator/tree/master/test/e2e/framework which might be a bit tricky to get into at the start, so no worries, just feel free to ask questions. 

This is the entry point of the e2e tests https://github.com/openshift/cluster-monitoring-operator/blob/master/test/e2e/main_test.go#L42 it does some pre-flight checks, and then just runs all the tests in parallel.

 Will comment on a test case to roughly give you an outline. But feel free to ask any questions! :)

Reply:
Thanks for dispelling my doubts  relatively ability to implement these tests and for very useful information. Will learn it asap and try to rewrite as minimum on case. Do not worry please about "initial confusion" it is way to do things simple and hope more reliable for me and a bit challenge :)